custom_components/tuya_local/devices/morento_hy4866_wf.yaml

```diff
@@ -0,0 +1,90 @@
+name: Air purifier
+products:
+  - id: 5a4rez30alfjfmdg
+    name: Morento HY4866-WF
+primary_entity:
+  entity: fan
+  translation_only_key: fan_with_presets
+  icon: "mdi:air-purifier"
+  dps:
+    - id: 1
+      name: switch
+      type: boolean
+    - id: 3
+      name: preset_mode
+      type: string
+      mapping:
+        - dps_val: sleep
+          value: sleep
+        - dps_val: Auto
+          value: auto
```

Comment:
For some reason fans don't have a predefined "auto" preset, but they have a "smart" preset which is similar. I don't remember the exact details, but I originally took the translated presets from a list that was in the HA fan implementation at the time, but they have since removed. There is also "strong" predefined, which could replace "very high".